pkg/model/estimator/local/ratio.go

```diff
@@ -102,12 +102,15 @@ func UpdateContainerEnergyByRatioPowerModel(containersMetrics map[string]*collec
 		// calculate the container gpu energy consumption
 		if accelerator.IsGPUCollectionSupported() {
 			containerResUsage = float64(container.CounterStats[config.GpuUsageMetric].Curr)
-			nodeTotalResUsage = nodeMetrics.GetNodeResUsagePerResType(config.GpuUsageMetric)
+			// the utilization based accounting already takes into account the container usage at the node level
+			nodeTotalResUsage = containerResUsage
```

Comment:
Would does this mean the ratio (containerResUsage/nodeTotalResUsage) always 1? 
Every container will have the same GPU energy = nodeResEnergyUtilization.

Reply:
I think instead of avoiding gpu metric in usage = 0.
We need to add the `config.GPUMemUtilization` in the feature list in the following function same as the other metric. https://github.com/sustainable-computing-io/kepler/blob/34473a4ba7d193af039e873915337ca2059a689e/pkg/collector/metric/utils.go#L25.

https://github.com/sustainable-computing-io/kepler/blob/34473a4ba7d193af039e873915337ca2059a689e/pkg/collector/metric/utils.go#L53

https://github.com/sustainable-computing-io/kepler/blob/f5a971fa2894f1fbe7de1d245ceea2d4701282ab/pkg/collector/metric/node_metric.go#L165-L175

Reply:
> Would does this mean the ratio (containerResUsage/nodeTotalResUsage) always 1?

@sunya-ch when the container doesn't use gpu, the containerResUsage is 0, so nodeTotalResUsage is 0 and this container only gets the idle power. When the container that uses GPU is counted, `containerResUsage/nodeTotalResUsage` is 1, then this container gets the dynamic power.

Reply:
I see. I think it looks good to me now. 
Also, even with the shared GPU case, things might be more complex with the utilization and cannot just treat as the same as the other metric.

Reply:
yes, GPU sharing will be a totally different solution, we probably will use similar [ML](http://users.ece.utexas.edu/~derek/Papers/HPCA2015_GPUPowerModel.pdf) based approach.

Reply:
Of course we can have different power models (ratio, ML, LR, etc).
But if the GPU partitioning can fully isolate applications, this might be vary similar to the container use case.